onnxruntime/core/providers/cpu/math/sign.cc

```diff
@@ -58,12 +58,13 @@ struct CallSignImpl {
 template <>
 struct CallSignImpl<MLFloat16> {
   void operator()(const Tensor* input, Tensor* output) const {
-    ConstEigenVectorMap<Eigen::half> input_data(
-        reinterpret_cast<const Eigen::half*>(input->Data<MLFloat16>()),
-        narrow<ptrdiff_t>(input->Shape().Size()));
-
-    EigenVectorMap<Eigen::half>(reinterpret_cast<Eigen::half*>(output->MutableData<MLFloat16>()),
-                                narrow<ptrdiff_t>(output->Shape().Size())) = input_data.array().cwiseSign();
+    auto span = input->DataAsSpan<MLFloat16>();
+    auto output_data = output->MutableData<MLFloat16>();
+    std::transform(span.begin(), span.end(), output_data, [](const MLFloat16& val) {
```

Comment:
Do we lose any vectorization by not using Eigen's cwiseSign?

Reply:
I could not find that code, and the table [here](https://eigen.tuxfamily.org/dox/group__CoeffwiseMathFunctions.html) has no mention of `cwiseSign`.

Reply:
Looks like it has been deprecated. Is this better? https://stackoverflow.com/a/34676254

Reply:
I do not think so. The nature of my change is that it avoids conversion to float altogether whereas with Eigen it converts it to float first.

Reply:
Not referring to the conversion as such. I'm more concerned about the application of the lambda (the actual operation) and if it makes sense for it to be vectorized using unaryExpr.